lib/slide_hub/cloud/engine/aws.rb

```diff
@@ -0,0 +1,280 @@
+          true
+        end
+
+        def self.delete_generated_files(key)
+          if key.empty?
+            return false
+          end
+          files = self.get_file_list(@config.image_bucket_name, key)
+          self.delete_files(@config.image_bucket_name, files)
+          true
+        end
+
+        def self.get_slide_download_url(key)
+          self.get_download_url(@config.bucket_name, key)
+        end
+
+        def self.delete_files(bucket, files)
+          Aws::S3::Client.new(region: @config.region).delete_objects({
+            bucket: bucket,
+            delete: {
+              objects: files,
+              quiet: true,
+            },
+          },) unless files.empty?
+        end
+
+        def self.get_download_url(bucket, key)
+          signer = Aws::S3::Presigner.new(client: Aws::S3::Client.new(region: @config.region))
+          url = signer.presigned_url(:get_object, bucket: bucket, key: key)
+          url
+        end
+
+        def self.create_policy
+          base_time = Time.zone.now.in_time_zone('UTC')
+
+          if !@config.aws_access_id.blank? && !@config.aws_secret_key.blank?
+            access_id = @config.aws_access_id
+            secret_key = @config.aws_secret_key
+            security_token = ''
+          else
+            ec2 = Aws::EC2::Client.new
+            credential = ec2.config[:credentials]
+            access_id = credential.access_key_id
+            secret_key = credential.secret_access_key
+            security_token = credential.session_token
+          end
+          region = @config.region
+          bucket_name = @config.bucket_name
+
+          self.populate_policy(base_time, access_id, secret_key, security_token, region, bucket_name)
+        end
+
+        def self.populate_policy(base_time, access_id, secret_key, security_token, region, bucket_name)
+          date_ymd = base_time.gmtime.strftime('%Y%m%d')
+          date_gm = base_time.gmtime.strftime('%Y%m%dT%H%M%SZ')
+          acl = 'public-read'
+          exp = base_time + 60 * 120
+          expires = exp.gmtime.strftime('%Y-%m-%dT%H:%M:%SZ')
+
+          #---------------------------------------------
+          # 1. Create a policy using UTF-8 encoding.
+          # This includes custom meta data named "x-amz-meta-title" for example
+          #---------------------------------------------
+          p_array = {
+            'expiration' => expires,
+            'conditions' => [
+              { 'bucket' => bucket_name },
+              ['starts-with', '$key', ''],
+              { 'acl' => acl },
+              { 'success_action_status' => '201' },
+              ['starts-with', '$Content-Type', 'application/octetstream'],
+              { 'x-amz-meta-uuid' => '14365123651274' },
+              ['starts-with', '$x-amz-meta-tag', ''],
+              { 'x-amz-credential' => "#{access_id}/#{date_ymd}/#{region}/s3/aws4_request" },
+            { 'x-amz-algorithm' => 'AWS4-HMAC-SHA256' },
```

Comment:
__[rubocop]__
* Style/AlignArray: Align the elements of an array literal if they span more than one line. [:ref](https://github.com/bbatsov/ruby-style-guide#align-multiline-arrays)